### ISSUE TITLE
npm audit messages

### ISSUE THREAD

**Issue by benben159** [pkp/pkp-lib#4951]
**Describe the bug**
`npm install` show vulnerabilities in node modules.

**To Reproduce**
1. clone stable-3_1_2 OJS branch
1. run `npm install` with npm version 6.4.1 and nodejs version 8.16.0, it shows this message at the end

```
... npm install messages ...
added 1127 packages from 642 contributors and audited 8113 packages in 30.666s
found 7 vulnerabilities (6 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```
`npm audit` shows suggestion to upgrade `css-loader` package to version 3.1.0 and `marked` package to 0.7.0.

**What application are you using?**
 OJS 3.1.2

**Additional information**
if this is irrelevant, you can close this issue and i'll post this to forum instead.

**Comment by NateWr** [comment]
Thanks for reporting @benben159!

I've pushed a couple of commits to fix this for the `master` branch: https://github.com/pkp/ojs/commit/b628631c9a5a516d9d8fac98f8889a11340fb960 and https://github.com/pkp/ui-library/commit/1581350d78986bbde0422f6bcd750f200deb510d.

I made an attempt at doing this for `stable-3_1_2`, but there's some outdated stuff there that is not updating cleanly. 

I'm going to leave the stable branch alone because these vulnerabilities are exclusively related to server-side functionality of node.js, which is not actually in use by any of our use-cases (we use it exclusively as a build process). The vulnerabilities aren't present just for building the static HTML/JS files to host the component library, which we do for the [ui-library in the docs](https://docs.pkp.sfu.ca/dev/ui-library/dev). They don't come into play for the built files in OJS/OMP. And they can't be triggered just by having the dependencies around in a `node_modules` directory, as far as I'm aware.

The only time these vulnerabilities come into play is when running the ui-library locally for development purposes, at which point if someone has compromised your machine then you've already got a problem.

Again, all vulnerability warnings are removed for the `master` branch, where we've significantly reduced our reliance on different dependencies. And moving forward we'll reduce these further by building on storybook or other third-party tooling for the component library itself.
